Makefile

```diff
@@ -1,36 +1,45 @@
 # Builds PEP files to HTML using sphinx
 
 PYTHON=python3
-VENV_DIR=venv
+VENVDIR=.venv
 JOBS=8
-RENDER_COMMAND=$(PYTHON) build.py -j $(JOBS)
+RENDER_COMMAND=$(VENVDIR)/bin/python3 build.py -j $(JOBS)
 
-render:
+render: venv
 	$(RENDER_COMMAND)
 
-pages: rss
+pages: venv rss
 	$(RENDER_COMMAND) --build-dirs
 
-fail-warning:
+fail-warning: venv
 	$(RENDER_COMMAND) --fail-on-warning
 
-check-links:
+check-links: venv
 	$(RENDER_COMMAND) --check-links
 
-rss:
-	$(PYTHON) generate_rss.py
+rss: venv
+	$(VENVDIR)/bin/python3 generate_rss.py
 
-clean:
+clean: clean-venv
 	-rm -rf build
 
-venv:
-	$(PYTHON) -m venv $(VENV_DIR)
-	./$(VENV_DIR)/bin/python -m pip install -r requirements.txt
-
-lint:
-	pre-commit --version > /dev/null || $(PYTHON) -m pip install pre-commit
-	pre-commit run --all-files
+clean-venv:
+	rm -rf $(VENVDIR)
 
-spellcheck:
-	pre-commit --version > /dev/null || $(PYTHON) -m pip install pre-commit
-	pre-commit run --all-files --hook-stage manual codespell
+venv:
+	@if [ -d $(VENVDIR) ] ; then \
+		echo "venv already exists."; \
+		echo "To recreate it, remove it first with \`make clean-venv'."; \
+	else \
+		$(PYTHON) -m venv $(VENVDIR); \
+		$(VENVDIR)/bin/python3 -m pip install -r requirements.txt; \
+		echo "The venv has been created in the $(VENVDIR) directory"; \
+	fi
```

Comment:
This sort of goes against the point of mke as I understand it -- if the name exists on the filesystem then `make` won't run the target, whereas we need a lot of logic here to do the same thing and support customisation of the `venv` directory.

I really don't want to overcomplicate this -- whilst I have a desire to make usability better for PEP authors, no-one complained up until now, and so we should balance usability to PEP authors for whom the current Makefile doesn't work and maintainability to ourselves.

A

Reply:
Yep, `make` is primarily for building/compiling files but this sort of thing is pretty common too.

This one is based on the CPython docs [`Makefile`](https://github.com/python/cpython/blob/434ffb7f1f86e6b0cdfad3ede59993934d86e464/Doc/Makefile#L152-L161): 

```makefile
venv:
	@if [ -d $(VENVDIR) ] ; then \
		echo "venv already exists."; \
		echo "To recreate it, remove it first with \`make clean-venv'."; \
	else \
		$(PYTHON) -m venv $(VENVDIR); \
		$(VENVDIR)/bin/python3 -m pip install -U pip setuptools; \
		$(VENVDIR)/bin/python3 -m pip install -r requirements.txt; \
		echo "The venv has been created in the $(VENVDIR) directory"; \
	fi
```

The devguide's [`Makefile`](https://github.com/python/devguide/blob/e14302a82ef4df599901eeab32578426295513ca/Makefile#L44-L47) is similar:

```makefile
venv:
	$(PYTHON) -m venv venv
	./venv/bin/python3 -m pip install --upgrade pip
	./venv/bin/python3 -m pip install -r requirements.txt
```

I'm sure I've seen similar in other https://github.com/python `Makefile`s, so using venvs is a fairy common pattern.

Reply:
I'm not a make expert, but as a general principle, I don't think we should let purity override practicality here, particularly when it comes to usability for PEP authors and contributors. We certainly want to reduce friction there as much as practical, and given all the changes lately and the remaining ones we want to do, making things easier rather than harder for users helps all that go more smoothly for everyone.